Initial Commit (#7242)
segmentio/segment-docs: src/connections/destinations/catalog/actions-s3/index.md

       "Action": "sts:AssumeRole",
       "Condition": {
         "StringEquals": {
           "sts:ExternalId": "<YOUR_EXTERNAL_ID>"
         }
       }
     }
   ]
 }
```
## Step 2: Add the AWS S3 (Actions) Destination in Segment
To finish the configuration, enable the AWS S3 (Actions) Destination in your workspace. 

1. Add the **AWS S3 (Actions)** destination from the Destinations tab of the catalog.
2. Select the data source you want to connect to the destination.
3. Provide a unique name for the destination.
4. Complete the destination settings:
   * Enter the name of the region in which the bucket you created above resides.
   * Enter the name of the bucket you created above. Be sure to enter the bucket's **name** and not URI.
   * Enter the ARN of the IAM role you created above. The ARN should follow the format `arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME.`
   * Enter the IAM External ID, which is a value set in the Trust Relationship under your AWS IAM Role.
5. Enable the destination.
6. Verify the Segment data is stored in the S3 bucket by navigating to the `<your_S3_bucket>/` in the AWS console. 

### Actions
Build your own Mappings. The following action is supported: 

> - [Sync to S3](#sync-to-s3)

Reviewer: @mayur-pitale sorry, one last thing, where is this linking to? there is no #syn-to-s3 section on this page

---

> title: S3 (Actions) Destination

Reviewer 2: ```suggestion
title: AWS S3 (Actions) Destination
```

This is the integration name on our backend. If you need to update this lmk

3. Enhanced Delivery Controls
The destination provides advanced options for batch size controls and file naming conventions. These controls can help optimize efficiency and simplify data retrieval workflows.

## Getting Started
Setting up the AWS S3 (Actions) destination is a straightforward process designed to help you quickly configure and deploy standardized event data to your Amazon S3 bucket. Follow these steps to get started:

### Prerequisites
Ensure you have the following in place before configuring the AWS S3 (Actions) destination:

- Amazon S3 Bucket: Create a bucket in your AWS account or use an existing one where you want to store the event data.
- AWS IAM Permissions: Verify that you have appropriate IAM roles with write access to the S3 bucket and permissions for the Segment connection.
- IAM Access IDs: Prepare your AWS IAM ARN ID and IAM External ID. These will be needed to authenticate and authorize Segment with your S3 bucket.

### Create an IAM role in the AWS console

> To setup the IAM role to properly authorize Segment with the AWS S3 (Actions) destination, follow the steps below. 

Reviewer: ```suggestion
To set up the IAM role to properly authorize Segment with the AWS S3 (Actions) destination:
```

> 1. Login to your AWS account 
> 2. Create a new or use an existing bucket with `PutObject`, `GetObject`, `ListObject` access to the S3 bucket.
> 3. Go to IAM > Roles > Create Role
> 4. Provide the following policy permissions for the IAM that was just created: 

Reviewer: ```suggestion
1. Log in to your AWS account. 
2. Create a new or use an existing bucket with `PutObject`, `GetObject`, `ListObject` access to the S3 bucket.
3. Navigate to **IAM > Roles > Create Role**.
4. Provide the following policy permissions for the IAM that was just created: 
```

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Sid": "PutObjectsInBucket",
            "Effect": "Allow",
            "Action": [
                "s3:PutObject",
                "s3:PutObjectAcl"
            ],
            "Resource": "arn:aws:s3:::<YOUR_BUCKET_NAME>/*"
        }
    ]
}
```

> 5. Click on the Trust Relationships tab and edit the trust policy to allow the IAM user to assume the role (If a user is not already created, please refer to the AWS documentation to create a user)

Reviewer: ```suggestion
5. Click on the Trust Relationships tab and edit the trust policy to allow the IAM user to assume the role. If a user is not already created, refer to the AWS documentation to create a user.
```

```json
{
   "Version": "2012-10-17",
   "Statement": [
     {
       "Sid": "",
       "Effect": "Allow",
       "Principal": {
         "AWS":                      
         	"arn:aws:iam::595280932656:role/customer-s3-prod-action-destination-access"
       },
       "Action": "sts:AssumeRole",
       "Condition": {
         "StringEquals": {
           "sts:ExternalId": "<YOUR_EXTERNAL_ID>"
         }
       }
     }
   ]
 }
```

> ## Add AWS S3 (Actions) Destination in Segment

Reviewer: ```suggestion
## Step 2: Add the AWS S3 (Actions) Destination in Segment
```

> To finish configuration, enable the AWS S3 (Actions) Destination in your workspace. 

Reviewer: ```suggestion
To finish the configuration, enable the AWS S3 (Actions) Destination in your workspace. 
```

> 1. Add the **AWS S3 (Actions)** destination from the Destinations tab of the catalog.
>     ![AWS S3](images/aws-s3-catalog.png)
> 2. Select the data source you'll connect to the destination.
> 3. Provide a unique name for the destination.
> 4. Complete the destination settings:
>    1. Enter the name of the region in which the bucket you created above resides.
>    2. Enter the name of the bucket you created above. Be sure to enter the bucket's **name** and not URI.
>    3. Enter the ARN of the IAM role you created above. The ARN should follow the format `arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME.`
>    4. Enter the IAM External ID, which is a value set in the Trust Relationship under your AWS IAM Role.
> 5. Enable the destination.
> 6. Verify Segment data is stored in the S3 bucket by navigating to the `<your_S3_bucket>/` in the AWS console. 

Reviewer: ```suggestion
1. Add the **AWS S3 (Actions)** destination from the Destinations tab of the catalog.
2. Select the data source you want to connect to the destination.
3. Provide a unique name for the destination.
4. Complete the destination settings:
   * Enter the name of the region in which the bucket you created above resides.
   * Enter the name of the bucket you created above. Be sure to enter the bucket's **name** and not URI.
   * Enter the ARN of the IAM role you created above. The ARN should follow the format `arn:aws:iam::ACCOUNT_ID:role/ROLE_NAME.`
   * Enter the IAM External ID, which is a value set in the Trust Relationship under your AWS IAM Role.
5. Enable the destination.
6. Verify the Segment data is stored in the S3 bucket by navigating to the `<your_S3_bucket>/` in the AWS console. 
```
